### PR TITLE
Export dataset

### PR DESCRIPTION
**Pull request by tkohr**
This PR adds components to download datasets in the different formats available.

TODOs:
- [x] Create UI components
- [ ] Adapt UI components to UI recommendations
- [ ] Handle events/logic of components
- [ ] Adapt mapping from elasticsearch
- [ ] Integrate components in datahub app

### CODE REVIEW

File: libs/util/shared/src/lib/models/search.model.ts [hunk 1]
```diff
@@ -33,6 +33,17 @@ export interface MetadataLink {
   url: string
 }
 
+// we will need to change model to contain format:
+// export type MetadataLinkFormats = 'OGC:WMS'|'OGC:WFS'|'OGC:WPS'|'ESRIFeature'|'csv'|'geosjon'|'json'|'shp'|'xls'|'xlsx'|'kml'|'gpkg'|'outbound'|'unknown'
+
+// export interface MetadataLink {
```
Comment (fgravin) [inline libs/util/shared/src/lib/models/search.model.ts:39]: why ?

Comment (jahow) [reply libs/util/shared/src/lib/models/search.model.ts:39]: this is something that we discussed, this model would be more appropriate for all kind of links, be it OGC services or simple files; the concept of "protocol" does not make sense for files, and the model currently does not contain a `title` (only a `name` which is more of a technical identifier)

**Comment by fgravin** [comment]
@tkohr 
tips when you are contributing:
run the following commands to avoid useless CI runs:
`npm run format`
`npm run affected:lint`

you can run `npm run affected:test` sometimes but it might be slow

Be sure to configure prettier so it automatically formats your code on the fly (plugins exist for IDEs)

**Comment by tkohr** [comment]
moved to #153

**Pull request closed without merging**